test/NonSilo.Tests/ClientBuilderTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans;
+using Orleans.Runtime.Configuration;
+using Xunit;
+
+namespace NonSilo.Tests
+{
+    /// <summary>
+    /// Tests for <see cref="ClientBuilder"/>.
+    /// </summary>
+    [TestCategory("BVT")]
+    public class ClientBuilderTests
+    {
+        /// <summary>
+        /// Tests that a client can be created without specifying configuration.
+        /// </summary>
+        [Fact]
+        public void ClientBuilder_NoSpecifiedConfigurationTest()
+        {
+            using (var client = new ClientBuilder().Build())
+            {
+                Assert.NotNull(client);
+            }
+        }
+
+        /// <summary>
+        /// Tests that a builder can not be used to build more than one client.
+        /// </summary>
+        [Fact]
+        public void ClientBuilder_DoubleBuildTest()
+        {
+            var builder = new ClientBuilder();
+            using (builder.Build())
+            {
+                Assert.Throws<InvalidOperationException>(() => builder.Build());
```

Comment:
is there a technical reason not to allow this or just philosophical? Because if the latter, maybe it is fine to re-use the same builder multiple times. If there is a technical why this doesn't work, then I don't mind for now